src/vaadin-grid-column.html

```diff
@@ -235,10 +235,10 @@
     }
 
     _renderHeaderAndFooter() {
-      if (this.headerRenderer) {
+      if (this.headerRenderer && this._headerCell) {
```

Comment:
Also considered having the check in
- vaadin-grid's `render()` but from the grid's point of view, calling `_renderHeaderAndFooter` should be safe regardless of the column's state.
- vaadin-grid-column's `__runRenderer()` but the function may expect the `cell` argument to be defined; it's the responsibility of the caller to make sure the arguments are valid. Otherwise, it wouldn't make sense to have a check for `this.headerRenderer` here either.